src/internal/Subscriber.ts

```diff
@@ -113,6 +113,39 @@ export class Subscriber<T> extends Subscription implements Observer<T> {
     }
   }
 
+  /**
+   * @ignore
+   * @internal
+   * Synchronous errors thrown within `subscribe` and caught by `_trySubscribe`
+   * can only be reported using a subscriber's `error` method if the subscriber
+   * is not stopped. This method traverses the `destination` chain of
+   * subscribers and uses the appropriate unhandled mechanism to report the
+   * error if a stopped subscriber is found.
+   */
+  reportError(err?: any): void {
+    let observer: PartialObserver<T> = this;
+    while (observer) {
+      if (observer instanceof Subscriber) {
+        const { destination } = observer;
+        if (observer.isStopped) {
```

Comment:
It seems like this could pass if someone just calls `error` twice in a row on Subscriber, no?

Reply:
The existing mechanism for guarding against re-enterant calls to `error` should not be compromised. `_reportError` is called only from within the `catch` inside `_trySubscribe`.

Even if a caller subverts the `private` qualification (added per request above) and calls `_reportError` instead of `error`, the `Subscriber` will still become stopped and further calls to either `_reportError` or `error` will not see the observer's error callback invoked.

Reply:
Wait. I'm wrong. The implementation needs to set `isStopped` to `true` for the subscriber at the root of the destination chain. I'll do that and add another test.

Reply:
I've updated the implementation so that `error` is called on the chain's root subscriber. This will see calls to `error` flow through the chain of subscribers and each will be marked as stopped.

Even if the subscriber is used as a destination for a subscriber that itself is not part of the chain, the observer's error callback will be invoked only once. So the existing mechanism for guarding against re-enterant calls to `error` should not be compromised.

`_reportError` is called only from within the `catch` inside `_trySubscribe`, but even if a caller subverts the `private` qualification (added per request above) and calls `_reportError` instead of `error`, all of the subscribers in the chain will still become stopped and further calls to either `_reportError` or `error` will not see the observer's error callback invoked.